security/aws/rds_unencrypted/AWS_Unencrypted_RDS_Instances.pt

```diff
@@ -0,0 +1,493 @@
+        "Version": "2014-10-31",
+        "ResourceName": dbInstanceArn,
+      },
+      headers: {
+        "Accept": "application/json"
+      }
+    }
+  EOS
+end
+
+#Process the response data, check for the tags and generate a list of unencrypted RDS instances
+script "js_rds_filter_map", type: "javascript" do
+  parameters "ds_rds_list_with_tags", "param_exclude_tags"
+  result "content"
+  code <<-EOS
+    var param_exclude_tags_lower=[];
+    for(var j=0; j < param_exclude_tags.length; j++){
+      param_exclude_tags_lower[j]=param_exclude_tags[j].toString().toLowerCase();
+    }
+
+    var content=[]
+    for(var i=0; i<ds_rds_list_with_tags.length; i++){
+      rds=ds_rds_list_with_tags[i]
+
+      //Check, if the tag present in entered param_exclude_tags, ignore the RDS instance if the tag matches/present.
+      var tags = rds['tags']
+      var isTagMatched=false
+      var tagKeyValue=""
+      for(var j=0; j < tags.length; j++){
+        tag = tags[j]
+        if((param_exclude_tags_lower.indexOf((tag['tagKey']).toLowerCase()) !== -1) || (param_exclude_tags_lower.indexOf((tag['tagKey']+'='+tag['tagValue']).toLowerCase()) !== -1)){
+          isTagMatched = true;
+        }
+        // Constructing tags with comma separated to display in detail_template
+        if((tag['tagValue']).length > 0){
+          tagKeyValue = tagKeyValue + ', '+ tag['tagKey']+'='+tag['tagValue']
+        }else{
+          tagKeyValue = tagKeyValue + ', '+ tag['tagKey']
+        }
+      }
+
+      var publicly_Accessible = rds['publicly_Accessible'];
+      if(publicly_Accessible === true){
+        publicly_Accessible = 'Yes';
+      }else{
+        publicly_Accessible = 'No';
+      }
+
+      var delete_Protection = rds['delete_Protection'];
+      if(delete_Protection === true){
+        delete_Protection = 'Yes';
+      }else{
+        delete_Protection = 'No';
+      }
+
+      //If the RDS instance tag does not match with entered param_exclude_tags, then check if RDS instance is Unencrypted.
+      if(!(isTagMatched) && (rds['storage_Encrypted'] === false)){
+        content.push({
+          dbInstance_Identifier: rds['dbInstance_Identifier'],
+          region: rds['region'],
+          availability_zone: rds['availability_zone'],
+          sec_availability_zone: rds['sec_availability_zone'],
+          publicly_Accessible: publicly_Accessible,
+          dbName: rds['dbName'],
+          db_instance_status: rds['db_instance_status'],
+          delete_Protection: delete_Protection,
+          tagKeyValue:(tagKeyValue.slice(2)),
+          dbInstanceArn: rds['dbInstanceArn'],
+          dbCluster_Identifier: rds['dbCluster_Identifier']
+        })
+      }
+    }
+  EOS
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "policy_rds_instance_list" do
+  validate $ds_rds_unencrypted_instance_list do
```

Comment:
Since this is a list its better to do validate_each on it.

Reply:
changed to 'validate_each'